src/request-handler.ts

```diff
@@ -165,6 +165,37 @@ export function createRequestHandler(
     }
   };
 
+  /**
+   * Sets the Content-Type header if needed based on the method and body.
+   *
+   * @param {HeadersInit} headers - The headers object where Content-Type will be set.
+   * @param {string} method - The HTTP method (e.g., GET, POST, PUT, DELETE).
+   * @param {any} [body] - Optional request body to determine if Content-Type is needed.
+   */
+  const setContentTypeIfNeeded = (
+    headers: HeadersInit,
+    method: string,
+    body?: any,
+  ) => {
+    // For PUT and DELETE methods, do not set Content-Type if no body is provided.
+    if (['PUT', 'DELETE'].includes(method) && !body) {
+      return;
+    }
+
+    // Automatically set Content-Type to 'application/json;charset=utf-8' if not already present.
+    if (headers instanceof Headers) {
+      if (!headers.has(CONTENT_TYPE)) {
```

Comment:
[nit-pick] I'd suggest merging this nested if condition into the upper one and just use `return;`. This way we merge ifs, and remove `else` so it should save some bytes 😄